NuRadioReco/modules/channelSinewaveSubtraction.py

```diff
@@ -20,21 +21,52 @@ class channelSinewaveSubtraction:
     def __init__(self):
         pass
 
-    def begin(self,  save_filtered_freqs=False, freq_band=(0.1, 0.7)):
+    def begin(self, save_filtered_freqs: bool = False, freq_band: tuple[float, float] = (0.1, 0.7)) -> None:
+        """
+        Initialize the CW filter module.
+
+        Parameters
+        ----------
+        save_filtered_freqs: bool (default: False)
+            Flag to save the identified noise frequencies for each channel.
+        freq_band: tuple (default: (0.1, 0.7))
+            Frequency band to calculate baseline RMS of fft spectrum. Used to identify noise peaks.
+            0.1 to 0.7 GHz is the default for RNO-G, based on bandpass. 
+
+        """
         self.save_filtered_freqs = [] if save_filtered_freqs else None
         self.freq_band = freq_band
 
-    def run(self, event, station, det=None, peak_prominence=4.0):
+    @register_run()
+    def run(self, event: 'NuRadioReco.framework.event.Event', station: 'NuRadioReco.framework.station.Station', det: 'NuRadioReco.detector.Detector' = None, peak_prominence: float = 4.0) -> None:
```

Comment:
@sjoerd-bouma can we use the backticks `` for typing?

Reply:
[This](https://peps.python.org/pep-0484/#forward-references) suggests that using a string instead of the actual class is acceptable only if the reference is resolvable once the module is loaded completely, which is not currently the case. I guess I would suggest explicitly importing the classes and removing the `'` altogether.

(backticks don't apply here, as that would be rst syntax that is resolved by sphinx when compiling the documentation, which is unfortunately completely separate from Python typing)

Reply:
But than the typing is correct and not the cause of the failure in the CI? Is that checked anyway?

Reply:
How do we fix that? `NuRadioReco.detector.detector.Detector` is not a class.